backend/src/SerialHandler.py

```diff
@@ -0,0 +1,261 @@
+# General imports =================================================================================
+import datetime
+import enum
+import multiprocessing as mp
+import os
+import sys
+import time
+import git
+import serial           # You'll need to run `pip install pyserial`
+from cobs import cobs   # pip install cobs
+import google.protobuf.message as Message
+
+# Project specific imports ========================================================================
+import proto.Python.CoreProto_pb2 as ProtoCore
+import proto.Python.TelemetryMessage_pb2 as TelemetryProto
+import proto.Python.ControlMessage_pb2 as ControlProto
+
+from src.support.Codec import Codec
+from src.support.ProtobufParser import ProtobufParser
+from src.support.CommonLogger import logger
+
+from src.Utils import Utils as utl, TEST_RECEIVE_SERIAL_MESSAGE, THREAD_MESSAGE_DB_WRITE, THREAD_MESSAGE_KILL, THREAD_MESSAGE_SERIAL_WRITE, WorkQ_Message
+
+# Constants ========================================================================================
+MIN_SERIAL_MESSAGE_LENGTH = 6
+
+UART_SERIAL_PORT = "/dev/ttyS0"
+RADIO_SERIAL_PORT = "/dev/ttyUSB0"
+
+# Data Classes =====================================================================================
+class SerialDevices(enum.Enum):
+    UART = 1
+    RADIO = 2
+
+
+# Class Definitions ===============================================================================
+class SerialHandler():
+    def __init__(self, thread_name: str, port: str, baudrate: int, message_handler_workq: mp.Queue):
+        """
+        This thread class creates threads to handle 
+        incoming and outgoing serial messages over 
+        the radio to the DMB and the uart to RCU.
+
+        Args:
+            thread_name (str):
+                The name of the thread.
+            port (str):
+                The port of the serial connection.
+            baudrate (int):
+                The baudrate of the serial connection.
+            message_handler_workq (mp.Queue):
+                The workq to send messages to the database handler.
+        """
+        logger.info(f"{thread_name} SerialHandler initializing")
+        self.port = port
+        self.baudrate = baudrate
+        self.thread_name = thread_name
+        self.send_message_workq = message_handler_workq
+        
+        # Open serial serial port
+        try:
+            self.serial_port = serial.Serial(port=port, baudrate=baudrate, bytesize=8, parity=serial.PARITY_NONE, timeout=None, stopbits=serial.STOPBITS_ONE)
+        except Exception:
+            self.serial_port = None
+            logger.error(f"Failed to open {thread_name} serial port {port}")
+            return
+        logger.success(f"Successfully opened {thread_name} serial port {port}")
```

Comment:
We could also potentially have a Statuses table instead of an error table to put a little green icon thing on things like "Serial Radio / Serial RCU" or "DB Thread"

Reply:
sounds good added a note about this too